### ISSUE TITLE
Trying to re-release old versions even with manifest config

### ISSUE THREAD

**Issue by cha0s** [google-github-actions/release-please-action#947]
### TL;DR

Trying to re-release old versions even with manifest config

### Expected behavior

The correct versions to be released

### Observed behavior

Please see: https://github.com/cha0s/flecks/pull/26/files

You can see right at the top is the existing manifest config.

You can also see that it is trying to create a "new" release for e.g. `packages/dox`: from version `4.0.10` to `4.0.3`(!!). It has many mistakes like this. No matter how many times I try to fix, it still does the wrong thing. I even added [another `bootstrap-sha`](https://github.com/cha0s/flecks/blob/f7868ee63a84760ede4be89e89391eef4e3dfae3/build/release-please-config.json#L2) to  avery recent commit and it still does the wrong thing.

### Action YAML

```yaml
on:
  push:
    branches:
      - master

name: release-please
jobs:
  release-please:
    runs-on: ubuntu-latest

    permissions:
      contents: write
      id-token: write
      pull-requests: write

    steps:

      - uses: GoogleCloudPlatform/release-please-action@v3
        id: release
        with:
          token: ${{secrets.FLECKS_GITHUB_TOKEN}}
          command: manifest
          config-file: build/release-please-config.json
          manifest-file: build/.release-please-manifest.json

      - if: ${{steps.release.outputs.releases_created}}
        uses: actions/checkout@v2

      - if: ${{steps.release.outputs.releases_created}}
        uses: actions/setup-node@v2
        with:
          cache: "npm"
          node-version: 18
          registry-url: "https://registry.npmjs.org"

      - if: ${{steps.release.outputs.releases_created}}
        env:
          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
        run: |
          npm ci
          npm run build
          npm run publish
```

### Log output

```text
It won't let me paste all the logs ("There was an error creating your issue: body is too long (maximum is 65536 characters).").

Here's a gist: https://gist.github.com/cha0s/67a0aa3cb4a5d3b27819d4ebd5f7fe9c
```

**Comment by cha0s** [comment]
Note: I can't wait for a resolution, so I manually limited the commit search which got it to do the right thing.

I believe this happened because `@flecks/headless` was a new package and did not have a release.

The bug is that it overwrote all other packages' release versions while searching for a (nonexistent) previous release of `@flecks/headless`.

**Comment by nicolas-vivot** [comment]
Same issue happening here.

For some unknown reason the release process start to use old (and already tagged) release versions, even though the version in the manifest configuration is there and correct.

![image](https://github.com/user-attachments/assets/6b65b533-6ea2-4140-b69e-765e3b960e98)

![image](https://github.com/user-attachments/assets/a6fff6cb-5bfe-411c-95ed-043e6cd6cb62)
